programs/move_loader_api/src/processor.rs

```diff
@@ -44,64 +45,37 @@ pub fn process_instruction(
 
     match command {
         LoaderInstruction::Write { offset, bytes } => {
-            MoveProcessor::do_write(keyed_accounts, offset, bytes)
+            MoveProcessor::do_write(keyed_accounts, offset, &bytes)
         }
         LoaderInstruction::Finalize => MoveProcessor::do_finalize(keyed_accounts),
         LoaderInstruction::InvokeMain { data } => {
-            MoveProcessor::do_invoke_main(keyed_accounts, data)
+            MoveProcessor::do_invoke_main(keyed_accounts, &data)
         }
     }
 }
 
 pub const PROGRAM_INDEX: usize = 0;
 pub const GENESIS_INDEX: usize = 1;
 
-// TODO: Not quite right yet
-/// Invoke information passed via the Invoke Instruction
-#[derive(Default, Debug, Serialize, Deserialize)]
-pub struct InvokeInfo {
-    /// Sender of the "transaction", the "sender" who is calling this program
-    pub sender_address: AccountAddress,
-    /// Name of the function to call
-    pub function_name: String,
-    /// Arguments to pass to the program being invoked
-    pub args: Vec<TransactionArgument>,
+/// Command to invoke
+#[derive(Debug, Serialize, Deserialize)]
+pub enum InvokeCommand {
```

Comment:
@garious Thoughts on this?  I think ideally we would want to have a new top-level instruction for creating the genesis account, on the same level as `LoaderInstruction`.

Reply:
Looks like it's getting a little weird for all the loaders to implement only `LoaderInstruction`. Instead, they should each have their own enum and implement a shared trait. The runtime can use the trait to create `InvokeMain` instructions. Beyond that, each program would be free to implement whatever additional instructions make sense for it.

Reply:
Yup, agreed.  @greg I'm planning to merge this when it passes CI and address the larger `LoaderInstruction` changes in a separate PR.  Any objections?